packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -911,7 +957,6 @@ export const DatePickerMixin = (subclass) =>
         } else {
           this.__keepInputValue = true;
           this.__commitDate(null);
-          this._selectedDate = null;
```

Comment:
**note:** This line is unnecessary since the selected date is already updated by `__commitDate(null)`.